### ISSUE TITLE
Out of bounds exception breaks GPU plugin after christmas update

### ISSUE THREAD

**Issue by remyvanlis** [runelite/runelite#18597]
### Type

Crash or hang

### Operating System

None

### OS Version

MacOS

### Bug description

https://github.com/runelite/runelite/blob/fc29d39ad4d52341da6748460959e099e0e28652/runelite-client/src/main/java/net/runelite/client/plugins/gpu/TextureManager.java#L256

This line is out of bounds when logging in with the GPU plugin enabled.

```
2024-12-10 13:29:19 CET [Client] ERROR injected-client - Client error
java.lang.ArrayIndexOutOfBoundsException: Index 256 out of bounds for length 256
 at net.runelite.client.plugins.gpu.TextureManager.computeTextureAnimations(TextureManager.java:256)
 at net.runelite.client.plugins.gpu.GpuPlugin.draw(GpuPlugin.java:1213)
 at bo.aj(bo.java:60059)
 at bo.tq(bo.java)
 at bo.run(bo.java:62715)
 at java.base/java.lang.Thread.run(Thread.java:833)
error_game_crash
```

suggestion: would changin line:226 from `float[] anims = new float[256 * 2];` to `float[] anims = new float[textures.length * 2 + 1];` solve this?

### RuneLite version

Runelite 1.10.46

### Logs

2024-12-10 13:29:19 CET [Client] ERROR injected-client - Client error
java.lang.ArrayIndexOutOfBoundsException: Index 256 out of bounds for length 256
 at net.runelite.client.plugins.gpu.TextureManager.computeTextureAnimations(TextureManager.java:256)
 at net.runelite.client.plugins.gpu.GpuPlugin.draw(GpuPlugin.java:1213)
 at bo.aj(bo.java:60059)
 at bo.tq(bo.java)
 at bo.run(bo.java:62715)
 at java.base/java.lang.Thread.run(Thread.java:833)
error_game_crash

**Comment by abextm** [comment]
fixed in d23c1e6ab98542f6e379b1ec8be5d7c73b817e45

**Comment by remyvanlis** [comment]
> It has to match the shader, which doesn't have access to textures.length

Thanks for the quick reply. That's fair! Also thanks for the quick fix.
